src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -57,6 +57,11 @@ static PSVersionInfo()
         {
             s_psVersionTable = new PSVersionHashTable(StringComparer.OrdinalIgnoreCase);
 
+            string assemblyPath = typeof(PSVersionInfo).GetTypeInfo().Assembly.Location;
+            FileVersionInfo fileVersionInfo = FileVersionInfo.GetVersionInfo(assemblyPath);
+
+            s_psV6Version = new SemanticVersion(fileVersionInfo.ProductVersion);
```

Comment:
The resulting `s_psV6Version` will look like this: `6.0.0-beta.7 SHA: d2fe36fb4137c2a7d6dbc0df22245d14256f3a74`.
I don't think the SHA or Commit info should be put in `$PSVersionTable.PSVersion`. I'm fine we change it from `6.0.0-beta` to `6.0.0-beta.7` though. How about split the information version string to get the `6.0.0-beta.7` part and create the SemanticVersion object with that?

Reply:
I think we could try to put the information in another attribute (not ProductVersion) to exclude parsing.

Reply:
Yes, that could be an option if we can find a reasonable attribute to store this info.
Splitting the information version string is fine by me too. And plus, we need to split that string anyway to reconstruct the value for `GitCommitId`.

Reply:
I don't  found a suitable attribute - we are limited by MSBuild's AL task.